contrib/endpoints/src/api_manager/check_security_rules.cc

```diff
@@ -41,29 +41,36 @@ const char kFailedFirebaseTest[] = "Failed to execute Firebase Test";
 const char kInvalidResponse[] = "Invalid JSON response from Firebase Service";
 const char kTestSuccess[] = "SUCCESS";
 
-void SetProtoValue(std::string key, ::google::protobuf::Value &value,
+const std::string GetFirebaseServer(context::RequestContext &context) {
+  return context.service_context()
+      ->config()
+      ->server_config()
+      ->api_check_security_rules_config()
+      .firebase_server();
+}
+
+void SetProtoValue(const std::string &key,
+                   const ::google::protobuf::Value &value,
                    ::google::protobuf::Value *head) {
   ::google::protobuf::Struct *s = head->mutable_struct_value();
   Map<std::string, google::protobuf::Value> *fields = s->mutable_fields();
   (*fields)[key] = value;
 }
 
-const std::string GetReleaseName(
-    std::shared_ptr<context::RequestContext> request_context) {
-  return request_context->service_context()->service_name() + ":" +
-         request_context->service_context()->service().apis(0).version();
+const std::string GetReleaseName(context::RequestContext &context) {
```

Comment:
I already tried that and I get the following compilation error. I think this is because the service_context() method is not marked as const. Let me know if you are OK with marking that method as const - I think it can be made into a const but I don't know why it was not marked const. 

contrib/endpoints/src/api_manager/check_security_rules.cc: In function 'const string google::api_manager::{anonymous}::GetReleaseName(const google::api_manager::context::RequestContext&)':
contrib/endpoints/src/api_manager/check_security_rules.cc:61:34: error: passing 'const google::api_manager::context::RequestContext' as 'this' argument of 'google::api_manager::context::ServiceContext* google::api_manager::context::RequestContext::service_context()' discards qualifiers [-fpermissive]
   return context.service_context()->service_name() + ":" +
                                  ^
contrib/endpoints/src/api_manager/check_security_rules.cc:62:34: error: passing 'const google::api_manager::context::RequestContext' as 'this' argument of 'google::api_manager::context::ServiceContext* google::api_manager::context::RequestContext::service_context()' discards qualifiers [-fpermissive]
          context.service_context()->service().apis(0).version();